eslint/eslint-patch/src/eslint-bulk-suppressions/generate-patched-file.ts

```diff
@@ -0,0 +1,280 @@
+    inputIndex = indexToScanTo;
+    return output;
+  }
+
+  let outputFile = '';
+
+  // Match this:
+  //    //------------------------------------------------------------------------------
+  //    // Requirements
+  //    //------------------------------------------------------------------------------
+  outputFile += scanUntilMarker('// Requirements');
+  outputFile += scanUntilMarker('//--');
+  outputFile += scanUntilNewline();
+
+  outputFile += `
+// --- BEGIN MONKEY PATCH ---
+const bulkSuppressionsPatch = require('../../bulk-suppressions-patch');
+const requireFromPathToLinterJS = bulkSuppressionsPatch.requireFromPathToLinterJS;
+`;
+
+  // Match this:
+  //    //------------------------------------------------------------------------------
+  //    // Typedefs
+  //    //------------------------------------------------------------------------------
+  const requireSection = scanUntilMarker('// Typedefs');
+
+  // Match something like this:
+  //
+  //    const path = require('path'),
+  //    eslintScope = require('eslint-scope'),
+  //    evk = require('eslint-visitor-keys'),
+  //
+  // Convert to something like this:
+  //
+  //    const path = require('path'),
+  //    eslintScope = requireFromPathToLinterJS('eslint-scope'),
+  //    evk = requireFromPathToLinterJS('eslint-visitor-keys'),
+  //
+  outputFile += requireSection.replace(/require\s*\((?:'([^']+)'|"([^"]+)")\)/g, (match, p1, p2) => {
+    const importPath = p1 ?? p2 ?? '';
+
+    if (importPath !== 'path') {
+      if (p1) {
+        return `requireFromPathToLinterJS('${p1}')`;
+      }
+      if (p2) {
+        return `requireFromPathToLinterJS("${p2}")`;
+      }
+    }
+
+    // Keep as-is
+    return match;
+  });
+  outputFile += `--- END MONKEY PATCH ---
+`;
+
+  // Match this:
+  //    const ruleContext = Object.freeze(
+  //      Object.assign(Object.create(sharedTraversalContext), {
+  //        id: ruleId,
+  //        options: getRuleOptions(configuredRules[ruleId]),
+  //        report(...args) {
+  //          /*
+  //           * Create a report translator lazily.
+  //
+  // Convert to something like this:
+  //
+  //    const ruleContext = Object.freeze(
+  //      Object.assign(Object.create(sharedTraversalContext), {
+  //        id: ruleId,
+  //        options: getRuleOptions(configuredRules[ruleId]),
+  //        report(...args) {
+  //          if (bulkSuppressionsPatch.shouldBulkSuppress({ filename, currentNode, ruleId })) return;
+  //          /*
+  //           * Create a report translator lazily.
+  //
+  outputFile += scanUntilMarker('const ruleContext = Object.freeze(');
+  outputFile += scanUntilMarker('report(...args) {');
+  outputFile += scanUntilNewline();
+  outputFile += `
+                        // --- BEGIN MONKEY PATCH ---
+                        if (bulkSuppressionsPatch.shouldBulkSuppress({ filename, currentNode, ruleId })) return;
+                        // --- END MONKEY PATCH ---
+`;
+
+  // TODO: MORE PATCHING GOES HERE
```

Comment:
Is this comment block still needed?  Or can we delete it?

Reply:
Good catch, will delete

Reply:
Done